titus_isolate/event/oversubscribe_event_handler.py

```diff
@@ -0,0 +1,280 @@
+        if not event[ACTION] == OVERSUBSCRIBE:
+            self.ignored_event(event, 'not a {} event'.format(OVERSUBSCRIBE))
+            return False
+
+        return True
+
+    # NOTE: this creates the CRD. should not be run and only really exists for testing purposes.
+    def __bootstrap(self):
+        try:
+            crd_list = self.__ext_api.list_custom_resource_definition(
+                field_selector=OPPORTUNISTIC_RESOURCE_FIELD_SELECTOR)
+            for item in crd_list.items:
+                log.debug('CRD already exists: %s', item.to_str())
+                return
+        except ApiException as e:
+            log.error('Exception when calling ApiextensionsV1beta1Api->list_custom_resource_definition: %s', e)
+
+        crd_version = V1beta1CustomResourceDefinitionVersion(name=OPPORTUNISTIC_RESOURCE_VERSION,
+                                                             served=True,
+                                                             storage=True)
+        crd_meta = V1ObjectMeta(name=OPPORTUNISTIC_RESOURCE_NAME)
+        crd_spec_names = V1beta1CustomResourceDefinitionNames(kind=OPPORTUNISTIC_RESOURCE_KIND,
+                                                              singular=OPPORTUNISTIC_RESOURCE_SINGULAR,
+                                                              plural=OPPORTUNISTIC_RESOURCE_PLURAL)
+        crd_spec = V1beta1CustomResourceDefinitionSpec(versions=[crd_version],
+                                                       group=OPPORTUNISTIC_RESOURCE_GROUP,
+                                                       names=crd_spec_names,
+                                                       scope='Namespaced')
+        crd = V1beta1CustomResourceDefinition(api_version=CRD_VERSION,
+                                              kind=CRD_KIND,
+                                              metadata=crd_meta,
+                                              spec=crd_spec)
+        try:
+            crd_resp = self.__ext_api.create_custom_resource_definition(crd)
+            log.debug('created CRD: %s', crd_resp.to_str())
+        except ApiException as e:
+            log.error('Exception when calling ApiextensionsV1beta1Api->create_custom_resource_definition: %s', e)
+
+    def __cleanup(self):
+        try:
+            oppo_list = self.__custom_api.list_namespaced_custom_object(version=OPPORTUNISTIC_RESOURCE_VERSION,
+                                                                        group=OPPORTUNISTIC_RESOURCE_GROUP,
+                                                                        plural=OPPORTUNISTIC_RESOURCE_PLURAL,
+                                                                        namespace=OPPORTUNISTIC_RESOURCE_NAMESPACE)
+            log.debug('cleanup: oppo list: %s', json.dumps(oppo_list))
+            clean_count = 0
+            for item in oppo_list['items']:
+                check_secs = self.__config_manager.get_float(OVERSUBSCRIBE_CLEANUP_AFTER_SECONDS_KEY,
```

Comment:
Probably a good idea to put a cut-out mechanism here.  If `OVERSUBSCRIBE_CLEANUP_AFTER_SECONDS_KEY` returns a `0` value for example we stop clean up altogether.

Reply:
Also we probably don't need to fetch this value inside the loop

Reply:
added the cut-out and moved outside of loop